src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -1011,26 +1011,90 @@ protected function send_plugin_theme_email( $type, $successful_updates, $failed_
 			$body[] = "\n";
 			$body[] = __( 'Please check your site now. It’s possible that everything is working. If there are updates available, you should update.' );
 			$body[] = "\n";
-
+            
+            // Check for failed themes and plugins listing with failed_update_plugins_themes option_name
+			$failed_update_plugins_themes = (array) get_site_option( 'failed_update_plugins_themes', array() );
+			
 			// List failed plugin updates.
 			if ( ! empty( $failed_updates['plugin'] ) ) {
 				$body[] = __( 'These plugins failed to update:' );
 
 				foreach ( $failed_updates['plugin'] as $item ) {
 					$body[] = "- {$item->name}";
+                    
+                // Failed Plugin list to store into the database 
```

Comment:
This block seems needlessly complex. You should be able to replace this whole block with something like this.
```suggestion
               $past_failures[ $item->item->plugin ] => time();
```
It needs to be placed in the right spot as well. Otherwise emails would never send. `wp_mail()` returns a boolean indicating if the email was successful. That can be checked before updating the option.

```
if ( $sent ) {
    update_option()
}
```

Reply:
How can we get the plugin name that are failed? $past_failures[ $item->item->plugin ] => time();

If we are going to change the multidimensional array then it may be issues with getting the plugin or theme name.

Reply:
Can you please explain me more about this code change? 

if ( $sent ) {
    update_option()
}

Reply:
// Failed Plugin list to store into the database 
$failed_plugin = array( 'name' => $item->name, 'failure_time' => time(), 'file_name' => $item->item->plugin );

This code help us to get the Plugin Name, Time stamp for past failure and file name for future to add in email.

Reply:
The plugin name is not needed. Instead, the plugin file or theme name should be used as the identifier.

Reply:
The `if( $sent )` is checking if the `wp_mail()` call is successful or not.